### PR TITLE
Add unit tests for Formalis::Coercion and fix existing coercions

### PR DESCRIPTION
**Pull request by JelF**
- Add unit tests for Formalis::Coercion
- Change existing coercions to handle edge cases from tests

### CODE REVIEW

File: lib/formalism/form/coercion.rb [hunk 1]
```diff
@@ -75,21 +75,19 @@ def to_float
...
+			Time.parse(@value) if @value && !@value.empty?
 		end
 
 		def to_boolean
 			@value && @value.to_s != 'false' ? true : false
 		end
 
 		def to_symbol
-			@value&.to_sym
+			@value.to_sym if @value.respond_to?(:to_sym)
```
Comment (AlexWayfer) [inline lib/formalism/form/coercion.rb]: I'm not sure that this is OK, not exception. There are two ways (options): suppress or raise. I think raising is OK until we have problems with this.

Comment (JelF) [reply lib/formalism/form/coercion.rb]: I'm not sure that forms should ever raise when valid from form data or json are parsed

I understand that validation errors will not be created here, but raise seems to be too brutal.
If you aren't sure too, i think better is to use my version

Comment (AlexWayfer) [reply lib/formalism/form/coercion.rb]: > when valid from form data

I don't think that something without `to_sym` method for `Symbol` field is valid.

> or json

It will have `to_sym` method.

Comment (AlexWayfer) [reply lib/formalism/form/coercion.rb]: I mean… 

You can't cover any type for any (business) case.

`Integer` doesn't have `to_sym` method, but theoretically it can be converted.

You don't know should it be converted or not for some case. Let users to decide.

Comment (JelF) [reply lib/formalism/form/coercion.rb]: > You don't know should it be converted or not for some case. Let users to decide.

It is a good idea, but how? By handling `NoMethodErrors`?

Comment (JelF) [reply lib/formalism/form/coercion.rb]: > I don't think that something without to_sym method for Symbol field is valid.

I mean valid as `application/x-www-form-urlencoded`, not valid as data for specific form.

Comment (JelF) [reply lib/formalism/form/coercion.rb]: > I don't think that something without to_sym method for Symbol field is valid.

Validation collects errors to a Set, not raises something

Comment (AlexWayfer) [reply lib/formalism/form/coercion.rb]: > It is a good idea, but how? By handling NoMethodErrors?

Maybe. Or fix the input. Or option for Formalism in the future if needed. We don't need for this now.

Comment (JelF) [reply lib/formalism/form/coercion.rb]: Ok, i will remove corresponding tests and revert changes

Comment (JelF) [reply lib/formalism/form/coercion.rb]: Done
